magefiles/rulesengine/repos/infra_deployments.go

```diff
@@ -1,17 +1,155 @@
+	Description: "Map Integration tests files when Integration component files are changed in the infra-deployments PR",
+	Condition: rulesengine.ConditionFunc(func(rctx *rulesengine.RuleCtx) bool {
+
+		return len(rctx.DiffFiles.FilterByDirGlob("components/integration/*")) != 0
+	}),
+	Actions: []rulesengine.Action{rulesengine.ActionFunc(func(rctx *rulesengine.RuleCtx) error {
+		AddLabelToLabelFilter(rctx, "integration-service")
+		return nil
+	})}}
+
+var InfraDeploymentsEnterpriseControllerTestFileChangeRule = rulesengine.Rule{Name: "Infra-deployments PR Enterprise Controller component File Change Rule",
+	Description: "Map Enterprise Controller tests files when EC component files are changed in the infra-deployments PR",
+	Condition: rulesengine.ConditionFunc(func(rctx *rulesengine.RuleCtx) bool {
+
+		return len(rctx.DiffFiles.FilterByDirGlob("components/enterprise-contract/*")) != 0
+	}),
+	Actions: []rulesengine.Action{rulesengine.ActionFunc(func(rctx *rulesengine.RuleCtx) error {
+		AddLabelToLabelFilter(rctx, "ec")
+		return nil
+	})}}
+
+var InfraDeploymentsJVMTestFileChangeRule = rulesengine.Rule{Name: "Infra-deployments PR Jvm-build-service component File Change Rule",
+	Description: "Map jvm-build-service tests files when Jvm-build-service component files are changed in the infra-deployments PR",
+	Condition: rulesengine.ConditionFunc(func(rctx *rulesengine.RuleCtx) bool {
+
+		return len(rctx.DiffFiles.FilterByDirGlob("components/jvm-build-service/*")) != 0
+	}),
+	Actions: []rulesengine.Action{rulesengine.ActionFunc(func(rctx *rulesengine.RuleCtx) error {
+		AddLabelToLabelFilter(rctx, "jvm-build-service")
+		return nil
+	})}}
+
+var InfraDeploymentsImageControllerTestFileChangeRule = rulesengine.Rule{Name: "Infra-deployments PR Image Controller component File Change Rule",
+	Description: "Map image-controller tests files when Image Controller component files are changed in the infra-deployments PR",
+	Condition: rulesengine.ConditionFunc(func(rctx *rulesengine.RuleCtx) bool {
+
+		return len(rctx.DiffFiles.FilterByDirGlob("components/image-controller/*")) != 0
+	}),
+	Actions: []rulesengine.Action{rulesengine.ActionFunc(func(rctx *rulesengine.RuleCtx) error {
+		AddLabelToLabelFilter(rctx, "image-controller")
+		return nil
+	})}}
+
+var InfraDeploymentsMultiPlatformTestFileChangeRule = rulesengine.Rule{Name: "Infra-deployments PR Multi Controller component File Change Rule",
+	Description: "Map multi platform tests files when Multi Controller component files are changed in the infra-deployments PR",
+	Condition: rulesengine.ConditionFunc(func(rctx *rulesengine.RuleCtx) bool {
+
+		return len(rctx.DiffFiles.FilterByDirGlob("components/multi-platform-controller/*")) != 0
+	}),
+	Actions: []rulesengine.Action{rulesengine.ActionFunc(func(rctx *rulesengine.RuleCtx) error {
+		AddLabelToLabelFilter(rctx, "multi-platform")
```

Comment:
similar as above, shall since multi-platform tests are skipped [here](https://github.com/konflux-ci/e2e-tests/blob/7eb9fd8575f3c52d76d4d2736f92ff6a5038e3bb/tests/build/multi-platform.go#L63), until that is enabled, no point in adding the labels, will it be good to skip running the jobs instead ?

Reply:
good point. although I think it's no harm to keep this rule here - it is at least guaranteed that we don't forget to add the rule once MP tests are enabled :)

should we then have a ticket for disabling the CI for changes to JBS/multi-platform component?

Reply:
for JBS tests skipping we got confirmation from konflux architect .. it will be reworked.. maybe tests needs to be rewritten as well.. they are yet to create stories in the development side .. and for the multi-platform tests being skipped due to [this issue](https://issues.redhat.com//browse/KFLUXBUGS-1179), until that gets fixed, I think we can use this story  for the skipping CI (openshift-ci job which runs e2e-tests)

Reply:
addressed in https://github.com/openshift/release/pull/55741, PTAL